### DISCUSSION TITLE
SB5.1 - OCI - Deploy/Kubernetes/Content failing

### DISCUSSION THREAD

**Discussion by ddevadat** [Sunbird-Ed/Community#323] [Installation] [answered]
Release : release-5.1.0
CSP: OCI
https://github.com/project-sunbird/sunbird-devops.git

`
TASK [sunbird-deploy : copy and template env file from swarm role] *************
failed: [localhost] (item=../../../../ansible/roles/stack-sunbird/templates/content-service_application.conf) => {
    "ansible_loop_var": "item",
    "changed": false,
    "item": "../../../../ansible/roles/stack-sunbird/templates/content-service_application.conf"
}

MSG:

AnsibleUndefinedVariable: 'cloudstorage_base_path' is undefined
`

**Comment by ddevadat** [comment]
in the ansible/roles/stack-sunbird/templates/content-service_application.conf

`
cloudstorage {
  metadata.replace_absolute_path={{ cloudstorage_replace_absolute_path | default('false') }}
  relative_path_prefix={{ cloudstorage_relative_path_prefix_content }}
  metadata.list={{ cloudstorage_metadata_list }}
  read_base_path="{{ cloudstorage_base_path }}"
  write_base_path={{ valid_cloudstorage_base_urls }}
}
`

Whats should be the value of
cloudstorage_base_path
valid_cloudstorage_base_urls
cloudstorage_replace_absolute_path

**Reply by beepdot** [reply]
@santhosh-tg @saiakhil46 Please provide your answers to this query.

**Comment by ddevadat** [comment]
any update on this?

**Reply by saiakhil46** [reply]
@ddevadat Iam sharing values for azure csp, you can update for OCI accordingly.
`cloud_storage_url: "https://{{ cloud_public_storage_accountname }}.blob.core.windows.net"`
`cloud_storage_base_url: "{{ cloud_storage_url }}"`
`cloudstorage_base_path: "{{ cloud_storage_url }}"`
`valid_cloudstorage_base_urls: '["{{ cloud_storage_url }}"]'`

**Reply by beepdot** [reply]
@saiakhil46 Can you also provide an example for S3 bucket for the `cloud_storage_url` var?

**Reply by saiakhil46** [reply]
@ddevadat Below is for AWS
`cloud_storage_url: "https://s3.{{ cloud_public_storage_region }}.amazonaws.com"`

**Comment by ddevadat** [comment, marked as answer]
Does cloud_storage_base_url and cloudstorage_base_path have the same value?
Going by the naming convention i was expecting cloudstorage_base_path, to be some kind of path. So just confirming it

**Reply by Krishnaj20** [reply]
@ddevadat - is your query resolved?

@gandham-santhosh

**Reply by saiakhil46** [reply]
@ddevadat It is the URL we are refering in `cloudstorage_base_path`
